Modules/_ctypes/callproc.c

```diff
@@ -2022,17 +2027,15 @@ create_pointer_type(PyObject *module, PyObject *cls)
                                        "_type_", cls);
         if (result == NULL)
             return result;
-        key = Py_NewRef(cls);
     } else {
         PyErr_SetString(PyExc_TypeError, "must be a ctypes type");
         return NULL;
     }
-    if (PyDict_SetItem(st->_ctypes_ptrtype_cache, key, result) < 0) {
-        Py_DECREF(result);
-        Py_DECREF(key);
-        return NULL;
+
+    if (info) {
+        info->pointer_type = Py_NewRef(result);
     }
```

Comment:
```suggestion
    assert(!info || info->pointer_type == result);
```
The registration could migrate to `PyCPointerType_init()` or `PyCPointerType_SetProto()`?

Reply:
To PyCPointerType_SetProto. It is now just checks presence of pointer_type and fails if so.

Reply:
Added extra checks.